src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/ComRuntimeHelpers.cs

```diff
@@ -0,0 +1,578 @@
+            if (pAttrs == IntPtr.Zero)
+            {
+                throw Error.CannotRetrieveTypeInformation();
+            }
+
+            try
+            {
+                return (ComTypes.TYPELIBATTR)Marshal.PtrToStructure(pAttrs, typeof(ComTypes.TYPELIBATTR));
+            }
+            finally
+            {
+                typeLib.ReleaseTLibAttr(pAttrs);
+            }
+        }
+
+        public static BoundDispEvent CreateComEvent(object rcw, Guid sourceIid, int dispid)
+        {
+            return new BoundDispEvent(rcw, sourceIid, dispid);
+        }
+
+        public static DispCallable CreateDispCallable(IDispatchComObject dispatch, ComMethodDesc method)
+        {
+            return new DispCallable(dispatch, method.Name, method.DispId);
+        }
+
+        internal static MethodInfo GetGetIDispatchForObjectMethod()
+        {
+            // GetIDispatchForObject always throws a PNSE in .NET Core, so we work around it by using GetComInterfaceForObject with our IDispatch type.
+            return typeof(Marshal).GetMethods().Single(m => m.Name == nameof(Marshal.GetComInterfaceForObject) && m.GetParameters().Length == 1 && m.ContainsGenericParameters).MakeGenericMethod(typeof(object), typeof(IDispatch));
+        }
+    }
+
+    /// <summary>
+    /// This class contains methods that either cannot be expressed in C#, or which require writing unsafe code.
+    /// Callers of these methods need to use them extremely carefully as incorrect use could cause GC-holes
+    /// and other problems.
+    /// </summary>
+    ///
+    internal static class UnsafeMethods
+    {
+        #region public members
+
+        public static unsafe IntPtr ConvertInt32ByrefToPtr(ref int value) { return (IntPtr)System.Runtime.CompilerServices.Unsafe.AsPointer(ref value); }
+        public static unsafe IntPtr ConvertVariantByrefToPtr(ref Variant value) { return (IntPtr)System.Runtime.CompilerServices.Unsafe.AsPointer(ref value); }
+
+        internal static Variant GetVariantForObject(object obj)
+        {
+            Variant variant = default;
+            if (obj == null)
+            {
+                return variant;
+            }
+            InitVariantForObject(obj, ref variant);
+            return variant;
+        }
+
+        internal static void InitVariantForObject(object obj, ref Variant variant)
+        {
+            Debug.Assert(obj != null);
+
+            // GetNativeVariantForObject is very expensive for values that marshal as VT_DISPATCH
+            // also is is extremely common scenario when object at hand is an RCW.
+            // Therefore we are going to test for IDispatch before defaulting to GetNativeVariantForObject.
+            if (obj is IDispatch)
+            {
+                variant.AsDispatch = obj;
+                return;
+            }
+
+            Marshal.GetNativeVariantForObject(obj, ConvertVariantByrefToPtr(ref variant));
+        }
+
+        [Obsolete("do not use this method", true)]
```

Comment:
These Obsolete attributes do not look useful. Are these methods accessed by reflection only? If yes, it should be enough to just add a comment. If not, delete them.

Reply:
Yep - accessed via reflection (seems like adding an Obsolete attribute like this was common in framework for parts of System.Dynamic). Will remove and switch to a comment.